modules/nw-autoscaling-ingress-controller.adoc

```diff
@@ -0,0 +1,222 @@
+// Module included in the following assemblies:
+//
+// * networking/ingress-controller-configuration.adoc
+
+:_content-type: PROCEDURE
+[id="nw-autoscaling-ingress-controller_{context}"]
+= Autoscaling an Ingress Controller
+
+Automatically scale an Ingress Controller to dynamically meet routing performance or availablility requirements such as the requirement to increase throughput. The following procedure provides an example for scaling up the default `IngressController`.
+
+.Prequisites
+. You installed the OpenShift CLI (`oc`).
+. You have access to the cluster as a user with the `cluster-admin` role.
+. You installed Custom Metrics Autoscaler Operator.
+
+.Procedure
+. Create a project in the `openshift-ingress-operator` namespace by running the following command:
++
+[source,terminal]
+----
+$ oc project openshift-ingress-operator
+----
+
+. Enable OpenShift monitoring for user-defined projects by applying the following config map:
++
+[source,terminal]
+----
+$ oc apply -f - <<EOF
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    enableUserWorkload: true
+EOF
+----
+
+. Create a Service Account to authenticate with Thanos by running the following command:
++
+[source,terminal]
+----
+$ oc create serviceaccount thanos && oc describe serviceaccount thanos
+----
++
+.Example output
+[source,terminal]
+----
+Name:                thanos
+Namespace:           openshift-ingress-operator
+Labels:              <none>
+Annotations:         <none>
+Image pull secrets:  thanos-dockercfg-b4l9s
+Mountable secrets:   thanos-dockercfg-b4l9s
+Tokens:              thanos-token-c422q
+Events:              <none>
+----
+
+. Define a `TriggerAuthentication` object within the `openshift-ingress-operator` namespace using the Service Account's token.
+
+.. Define the variable `$secret` that contains the secret by running the following command:
++
+[source,terminal]
+----
+$ secret=$(oc get secret | grep thanos-token | head -n 1 | awk '{ print $1 }')
+----
+
+.. Create the `TriggerAuthentication` object and pass the value of `$secret` to the `TOKEN` karg:
++
+[source,terminal]
+----
+$ oc process TOKEN="$secret" -f - <<EOF | oc apply -f -
+apiVersion: template.openshift.io/v1
+kind: Template
+parameters:
+- name: TOKEN
+objects:
+- apiVersion: keda.sh/v1alpha1
+  kind: TriggerAuthentication
+  metadata:
+    name: keda-trigger-auth-prometheus
+  spec:
+    secretTargetRef:
+    - parameter: bearerToken
+      name: \${TOKEN}
+      key: token
+    - parameter: ca
+      name: \${TOKEN}
+      key: ca.crt
+EOF
+----
+
+. Create a role for reading metrics from Thanos by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f - <<EOF
+apiVersion: rbac.authorization.k8s.io/v1
```

Comment:
is this the output of running this command `$ oc apply -f - <<EOF`?

Reply:
No, it's another instance of passing a config using the interactive shell. I've separated it out for simplicity though